### PR TITLE
Payment Channel: Sign data should include channel address

### LINKED ISSUE #612: PaymentChannel: Vouchers can be replayed across channels
`UpdateChannelState` allows either party of a PaymentChannel to submit a voucher signed by their counterparty in order to redeem channel funds. 

Signed vouchers do not include any unique information to link the voucher to a specific channel between the two parties, meaning that a voucher signed by either of a channel's parties may be submitted to any channel that party is a member of.

In order to ensure signed vouchers only apply to a single channel, the address of the PaymentChannel actor should be included in the signed payload. When validating a submitted signature, `UpdateChannelState` should check that this address matches `Message().Receiver()`.

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Closes #612.

### CODE REVIEW

**Comment by wadeAlexC** [comment]
LGTM on the `paych_actor` side, though I don't understand the cbor_gen parts!

File: actors/builtin/paych/paych_actor.go [hunk 1]
```diff
@@ -93,6 +93,8 @@ type UpdateChannelStateParams struct {
 // A voucher is sent by `From` to `To` off-chain in order to enable
 // `To` to redeem payments on-chain in the future
 type SignedVoucher struct {
+	// PaymentChannelAddr is the address of the payment channel this signed voucher is valid for
+	PaymentChannelAddr addr.Address
```
Comment (anorth) [inline actors/builtin/paych/paych_actor.go:97]: Let's just call it `ChannelAddr`, I think "payment" is sufficiently implied (remember to regenerate cbor).

File: actors/builtin/paych/paych_actor.go [hunk 2]
```diff
@@ -157,6 +159,11 @@ func (pca Actor) UpdateChannelState(rt vmr.Runtime, params *UpdateChannelStatePa
 		rt.Abortf(exitcode.ErrIllegalArgument, "voucher signature invalid: %s", err)
 	}
 
+	pchAddr := rt.Message().Receiver()
+	if pchAddr != sv.PaymentChannelAddr {
+		rt.Abortf(exitcode.ErrIllegalArgument, "voucher payment channel address does not match receiver")
```
Comment (anorth) [inline actors/builtin/paych/paych_actor.go:164]: Include `sv.PaymentChannelAddr` in the message

File: actors/builtin/paych/paych_test.go [hunk 3]
```diff
@@ -124,41 +124,47 @@ func TestPaymentChannelActor_CreateLane(t *testing.T) {
...
 		sig            *crypto.Signature
 		verifySig      bool
 		expExitCode    exitcode.ExitCode
 	}{
 		{desc: "succeeds", targetCode: builtin.AccountActorCodeID,
-			amt: 1, epoch: 1, tlmin: 1, tlmax: 0,
+			amt: 1, paymentChannel: paychAddr, epoch: 1, tlmin: 1, tlmax: 0,
 			sig: sig, verifySig: true,
 			expExitCode: exitcode.Ok},
+		{desc: "succeeds", targetCode: builtin.AccountActorCodeID,
```
Comment (anorth) [inline actors/builtin/paych/paych_test.go:138]: Desc should be "fails if channel address doesn't match"

File: actors/builtin/paych/paych_actor.go (hunk 1 above)
Comment (aarshkshah1992) [reply actors/builtin/paych/paych_actor.go:97]: This is done.

Comment (aarshkshah1992) [reply actors/builtin/paych/paych_actor.go]: This is done.

File: actors/builtin/paych/paych_actor.go (hunk 2 above)
Comment (aarshkshah1992) [reply actors/builtin/paych/paych_actor.go]: This is done.

File: actors/builtin/paych/paych_test.go (hunk 3 above)
Comment (aarshkshah1992) [reply actors/builtin/paych/paych_test.go]: This is done.

**Pull request merged**